lib/rbnacl/sodium.rb

```diff
@@ -29,14 +29,19 @@ def primitive
       sodium_primitive
     end
 
-    def sodium_constant(constant, name = constant)
-      fn = if sodium_primitive
-             "crypto_#{sodium_type}_#{sodium_primitive}_#{constant.to_s.downcase}"
-           else
-             "crypto_#{sodium_type}_#{constant.to_s.downcase}"
-           end
-      attach_function fn, [], :size_t
-      const_set(name, public_send(fn))
+    def sodium_constant(constant, name: constant, fallback: nil)
+      fn_name_components = ["crypto", sodium_type, sodium_primitive, constant.to_s.downcase]
+      fn_name = fn_name_components.reject(&:nil?).join("_")
```

Comment:
Super minor nit: `reject(&:nill?)` can be replaced with `compact` (surprised RuboCop doesn't lint that?)